src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -2041,44 +2041,83 @@ private BoundExpression BindArgumentValue(DiagnosticBag diagnostics, ArgumentSyn
             }
 
             var declarationExpression = (DeclarationExpressionSyntax)argumentSyntax.Expression;
-            var declaration = (TypedVariableComponentSyntax)declarationExpression.VariableComponent;
-            var typeSyntax = declaration.Type;
+            var typeSyntax = declarationExpression.Type();
+            bool isVar;
+
+            // Is this a local?
+            SourceLocalSymbol localSymbol = this.LookupLocal(declarationExpression.Identifier());
 
-            if (InConstructorInitializer || InFieldInitializer)
+            if ((object)localSymbol != null)
             {
-                Error(diagnostics, ErrorCode.ERR_ExpressionVariableInConstructorOrFieldInitializer, declarationExpression);
-            }
+                if (InConstructorInitializer || InFieldInitializer)
```

Comment:
This is intentional, because the variables are fields rather than locals and there is no issue with local scope.